Club Pages (all clubs, club pages)

Jchuang7 opened the pull request:
Added dynamic routing to each club page, test functionalities since there's no data to test yet.

Jchuang7 committed:
Club Pages Formatting

Basic formatting for club pages, with information to be inputted

Jchuang7 committed:
Removed unnecessary component

removed component

Jchuang7 committed:
Dynamic Club Pages

Added dynamic club pages when opening from the all clubs list and basic functionalities.

varangill commented on client/src/App.tsx:
```diff
@@ -9,3 +9,32 @@ ReactDOM.createRoot(document.getElementById("root")!).render(
+  text: string;
+}
+
+function createUL(listItems: Listitem[]): void {
```
React has custom ways to create a component for objects within an array by calling `map` on the array that we should use, it's a lot cleaner and avoids raw JS DOM manipulation. @kravich2 's [recent PR here](https://github.com/varangill/ClubHub/pull/9/files#diff-26cee6f83578cbe23a73e0f3c317f0764a2a250bc18cc362c69436f42c804730R67) has an example of it

varangill replied:
Also this sort of page-specific logic shouldn't be in `App.tsx`

varangill commented on client/src/Club.css:
```diff
@@ -0,0 +1,28 @@
+  .custom-header {
+    background-color: #333;
+    color: white;
+    padding: 10px; /* Add padding to the entire header */
```
Not a blocker but we don't really need comments on CSS properties (unless it's something really weird/confusing), you don't have to remove them here, just a head's up for the future to save some time

varangill reviewed (changes requested):
Comment on the createUL function, let us know if you need help

Also could you post an image in the PR itself for each page that's created/updated in the PR?

Jchuang7 committed:
Removed code from App.tsx

I think I was just testing something in here, the code wasn't used so I removed it.

Jchuang7 commented:
<img width="1114" alt="image" src="https://github.com/varangill/ClubHub/assets/99681830/9c4ac817-2514-48d7-a10c-d35fe743198c">
<img width="1619" alt="Screen Shot 2024-01-09 at 1 49 07 PM" src="https://github.com/varangill/ClubHub/assets/99681830/2b510fc9-9a9b-4ac9-a5c8-e326ae4022f4">
<img width="962" alt="image" src="https://github.com/varangill/ClubHub/assets/99681830/4ab7adc4-eaf8-40e6-91a0-b1472cb5e15f">
<img width="1620" alt="image" src="https://github.com/varangill/ClubHub/assets/99681830/e5960224-70eb-4365-af3a-7ee4300086f2">

I didn't set up example data yet so the information for members and announcments are the same for now.

Jchuang7 committed:
Changes to all clubs functionalities

Added Keerthen's search clubs functionality to the all clubs page. Also added and changed the example data for clubs.

kravich2 committed:
Front End of myclub page

added some more front end details and designs to Justin's work

Jchuang7 committed:
Added padding to search bar and testing global variables

Made changes to padding on search bar and testing on global variables

Jchuang7 committed:
Added my clubs data retrieval

Retrieved club membership info on specific user from back end and populated my club page with that info.

Jchuang7 committed:
Added join club feature

Added the join club feature on club pages and has it show up in my clubs after.

Jchuang7 committed:
Removed unnecessary files and minor changes

Removed files and made changes to post call and id for joined clubs
